examples/010_pymechanical_shell_workflow.py

```diff
@@ -0,0 +1,327 @@
+
+from concurrent.futures import ThreadPoolExecutor
+import pathlib
+import tempfile
+import textwrap
+
+# %%
+# Import PyACP, PyMechanical, and PyDPF Composites.
+
+# isort: off
+import ansys.acp.core as pyacp
+import ansys.dpf.composites as pydpf_composites
+import ansys.mechanical.core as pymechanical
+
+# sphinx_gallery_thumbnail_path = '_static/gallery_thumbnails/sphx_glr_010_pymechanical_shell_workflow_thumb.png'
+
+# %%
+# Start the ACP, Mechanical, and DPF servers. We use a ``ThreadPoolExecutor``
+# to start them in parallel.
+with ThreadPoolExecutor() as executor:
+    futures = [
+        executor.submit(pymechanical.launch_mechanical, batch=True),
+        executor.submit(pyacp.launch_acp),
+        executor.submit(pydpf_composites.server_helpers.connect_to_or_start_server),
+    ]
+    mechanical, acp, dpf = (fut.result() for fut in futures)
+
+# %%
+# Get example input files
+# -----------------------
+#
+# Create a temporary working directory, and download the example input files
+# to this directory.
+
+working_dir = tempfile.TemporaryDirectory()
+working_dir_path = pathlib.Path(working_dir.name)
+input_geometry = pyacp.example_helpers.get_example_file(
+    pyacp.example_helpers.ExampleKeys.CLASS40_AGDB, working_dir_path
+)
+
+# %%
+# Generate the mesh in PyMechanical
+# ---------------------------------
+#
+# Load the geometry into Mechanical, generate the mesh, and export it to the
+# appropriate transfer format for ACP.
+
+mesh_path = working_dir_path / "mesh.h5"
+mechanical.run_python_script(
+    # This script runs in the Mechanical Python environment, which uses IronPython 2.7.
+    textwrap.dedent(
+        f"""\
+        geometry_import = Model.GeometryImportGroup.AddGeometryImport()
+
+        import_format = Ansys.Mechanical.DataModel.Enums.GeometryImportPreference.Format.Automatic
+        import_preferences = Ansys.ACT.Mechanical.Utilities.GeometryImportPreferences()
+        import_preferences.ProcessNamedSelections = True
+        import_preferences.ProcessCoordinateSystems = True
+
+        geometry_file = {str(input_geometry)!r}
+        geometry_import.Import(
+            geometry_file,
+            import_format,
+            import_preferences
+        )
+
+        for body in Model.Geometry.GetChildren(
+            Ansys.Mechanical.DataModel.Enums.DataModelObjectCategory.Body, True
+        ):
+            body.Thickness = Quantity(1e-6, "m")
+
+        Model.Mesh.GenerateMesh()
+        """
+    )
+)
+pyacp.mechanical_integration_helpers.export_mesh_for_acp(mechanical=mechanical, path=mesh_path)
+
+# %%
+# Set up the ACP model
+# --------------------
+#
+# Setup basic ACP lay-up based on the mesh in ``mesh_path``, and export material and composite
+# definition file to output_path.
+
+composite_definitions_h5 = "ACPCompositeDefinitions.h5"
+matml_file = "materials.xml"  # TODO: load an example materials XML file instead of defining the materials in ACP
```

Comment:
Not relevant for this PR but would it be possible to load materials in pymechanical from ENGD (WB material DB) and generate the mat.xml?

Reply:
Do you mean the file `ENGD/EngineeringData.xml`?

For import, `Model.Materials.Import` seems to work (but only if all existing materials are deleted first).

At first glance, I couldn't find a method to export from Mechanical (other than CDB), though.